compiler/src/dotty/tools/dotc/typer/EtaExpansion.scala

```diff
@@ -157,6 +157,30 @@ class LiftComplex extends Lifter {
 }
 object LiftComplex extends LiftComplex
 
+/** Lift complex + lift the prefixes */
+object LiftCoverage extends LiftComplex {
+
+  var liftEverything = false
```

Comment:
```suggestion
  private var liftEverything = false
```
This global state seems a bit dangerous. Maybe there is a better way to do it.

Reply:
One way is to put this information in the context. It can be done as follows
```scala
  private val LiftEverything = new Property.Key[Boolean]
  private def liftEverything(using Context): Int =
    ctx.property(LiftEverything).contains(true)
  private def liftEverythingContext(using Context): Context =
    ctx.fresh.setProperty(LiftEverything, true)
```

```scala
  def liftForCoverage(defs: mutable.ListBuffer[tpd.Tree], tree: tpd.Apply)(using Context) = {
    val liftedFun = liftApp(defs, tree.fun)
    val liftedArgs = liftArgs(defs, tree.fun.tpe, tree.args)(using liftEverythingContext)
    tpd.cpy.Apply(tree)(liftedFun, liftedArgs)
  }
```